src/Illuminate/Queue/Worker.php

```diff
@@ -499,7 +508,17 @@ protected function raiseFailedJobEvent($connectionName, $job, $e)
      */
     protected function queueShouldRestart($lastRestart)
     {
-        return $this->getTimestampOfLastQueueRestart() != $lastRestart;
+        return $this->hasRunEnoughJobs() || $this->getTimestampOfLastQueueRestart() != $lastRestart;
+    }
+
+    /**
+     * Determine if the queue worker has run enough jobs.
+     *
+     * @return bool
+     */
+    protected function hasRunEnoughJobs()
+    {
+        return $this->options->jobs > 0 ? $this->jobs >= $this->options->jobs : false;
```

Comment:
Shouldnt this be `$this->options->maxJobs`?

Reply:
Thanks.